docs/courses/robotics/3.5-computer-vision.md

```diff
@@ -0,0 +1,157 @@
+
+---
+
+### AprilTags
+
+Next, you will utilize **AprilTags**, a popular fiducial marker system for reliable and robust spatial referencing and object identification. AprilTags are often used in robotics for locating objects and navigating environments.
+
+#### Bill of Materials
+
+- [AprilTags Python Library](https://pypi.org/project/apriltag/)
+- [AprilTags Documentation](https://april.eecs.umich.edu/software/apriltag.html)
+- [Raspberry Pi Camera Module](https://www.raspberrypi.org/products/camera-module-v2/)
+- [Motorized Microscope]
+#### Demo
+
+✅ Watch [Vision Programming with AprilTags](https://youtu.be/TG9KAa2EGzQ)
+
+The following code demonstrates how to use **AprilTags** for spatial referencing by detecting tags and performing ID lookup.
+
+```python
+import apriltag
+import cv2
+
+# Load an image with AprilTags
+img = cv2.imread('apriltag_image.jpg')
+gray_image = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+
+# Initialize AprilTag detector
+detector = apriltag.Detector()
+
+# Detect tags
+tags = detector.detect(gray_image)
+
+# Display tag information
+for tag in tags:
+    print(f"Detected tag ID: {tag.tag_id}")
+    cv2.rectangle(img, 
+                  (int(tag.corners[0][0]), int(tag.corners[0][1])), 
+                  (int(tag.corners[2][0]), int(tag.corners[2][1])), 
+                  (0, 255, 0), 2)
+
+cv2.imshow('AprilTag Detected', img)
+cv2.waitKey(0)
+cv2.destroyAllWindows()
+```
+
+In this demo, you will detect **AprilTags** in an image and draw bounding boxes around them. You can then use the tag ID for further referencing tasks.
+
+---
+
+### Motorized Microscope and OpenCV
+
+After detecting regions of interest (ROI) using **OpenCV** and **AprilTags**, you will integrate a motorized microscope. The microscope will move automatically based on the ROI found in the image, providing fine control and automation for analyzing samples.
+
+#### Demo
+
+✅ Watch [Controlling a Motorized Microscope](https://youtu.be/fHZSsAKThW4)
+
+In this task, you will automate the movement of a motorized microscope using **OpenCV**-detected ROIs. Below is a conceptual pseudo-code for moving the microscope:
+
+```python
+def move_microscope_to_roi(roi):
+    # Command to move the motorized microscope to ROI based on coordinates
+    microscope.move_to(roi['x'], roi['y'])
+
+# Sample ROI detected using OpenCV (this would be dynamically detected)
+roi = {'x': 150, 'y': 200}
+
+# Move the microscope to the detected ROI
+move_microscope_to_roi(roi)
+```
+
+You can integrate the motorized microscope with your **OpenCV** detection pipeline, enabling real-time analysis and precise control.
+
+---
+
+## 🚀 Quiz
+
+::::{tab-set}
+:sync-group: category
+
+:::{tab-item} Sp/Su 2024
+:sync: sp2024
+
+[Quiz URL]
+:::
+
+::::
+
+---
+
+## 📄 Assignment
+
+Create a script that uses **OpenCV** and **AprilTags** to detect regions of interest in a sample and control a motorized microscope to move to those regions.
```

Comment:
This could be for imaging particles (particle size distribution), searching for an object/item within a region (blank elsewhere), or a metal or biological sample.

https://github.com/AccelerationConsortium/ac-training-lab/issues/28